stats/dummy/collector.go

```diff
@@ -69,3 +69,8 @@ func (c *Collector) Link() string {
 func (c *Collector) GetRequiredSystemTags() lib.TagSet {
 	return lib.TagSet{} // There are no required tags for this collector
 }
+
+// IsRunning returns the current status of the collector
+func (c *Collector) IsRunning() bool {
+	return c.running
```

Comment:
This will cause a race condition since it doesn't lock (see `Run()`), but that's beside the point. Generally in Go it's not a very good idea to have methods like this, because the only way they can be used is with continuous polling, like you've done with the `time.Sleep(time.Millisecond)`, and it's also easier to introduce race conditions. It's much, much better to use channels or something like `sync.WaitGroup` (or even `sync.Cond` for particularly tricky cases) for this type of synchronization.